lib/dal/src/pkg/export.rs

```diff
@@ -178,27 +165,25 @@ impl PkgExporter {
                 }
             }
 
-            if !is_deleted {
-                let variant_spec = self.export_variant(ctx, change_set_pk, variant).await?;
-                self.variant_map.insert(
-                    change_set_pk.unwrap_or(ChangeSetPk::NONE),
-                    *variant.id(),
-                    variant_spec.to_owned(),
-                );
-                if variant_spec.unique_id.is_some() {
-                    if let Some(default_variant_id) = default_variant_id {
-                        if variant.id() == &default_variant_id {
-                            default_variant_unique_id = variant_spec.unique_id.to_owned();
-                        }
+            let variant_spec = self.export_variant(ctx, change_set_pk, variant).await?;
+            self.variant_map.insert(
+                change_set_pk.unwrap_or(ChangeSetPk::NONE),
+                *variant.id(),
+                variant_spec.to_owned(),
+            );
+            if variant_spec.unique_id.is_some() {
+                if let Some(default_variant_id) = default_variant_id {
+                    if variant.id() == &default_variant_id {
+                        default_variant_unique_id = variant_spec.unique_id.to_owned();
                     }
                 }
-                schema_spec_builder.variant(variant_spec);
             }
+            schema_spec_builder.variant(variant_spec);
         }
 
-        if in_change_set && is_deleted {
+        if is_deleted {
             schema_spec_builder.deleted(true);
-        } else if in_change_set {
```

Comment:
This means we will export all schemas into this change set in a workspace export whether they're only present on head or in the change set. I have to check the import logic but at the very least it's stuffing a ton of data into the export that doesn't need to be there.

Reply:
Hmm, I need to test this, you are right on this. The problem is that sometimes we need to export the schema in the change-set and sometimes we don't.